src/south/ProtocolHandler.class.js

```diff
@@ -35,20 +37,70 @@ class ProtocolHandler {
    * @constructor
    * @param {*} dataSource - The data source
    * @param {Engine} engine - The engine
+   * @param {object} supportedModes - The supported modes
    * @return {void}
    */
-  constructor(dataSource, engine) {
+  constructor(dataSource, engine, supportedModes) {
     this.dataSource = dataSource
     this.engine = engine
     this.encryptionService = EncryptionService.getInstance()
     this.compressionLevel = 9
     const { engineConfig } = this.engine.configService.getConfig()
     this.engineConfig = engineConfig
+    this.supportedModes = supportedModes
 
     const { logParameters } = this.dataSource
     this.logger = new Logger()
     this.logger.changeParameters(this.engineConfig, logParameters)
 
+    if (this.supportedModes) {
```

Comment:
should we make a function checkSupportedModes()

Reply:
Fixed